connection_configuration.go

```diff
@@ -0,0 +1,323 @@
+		cfg.ValidateDefaultParameters, parsingErr = parseConfigBool(value)
+	case "clientrequestmfatoken":
+		cfg.ClientRequestMfaToken, parsingErr = parseConfigBool(value)
+	case "clientstoretemporarycredential":
+		cfg.ClientStoreTemporaryCredential, parsingErr = parseConfigBool(value)
+	case "tracing":
+		cfg.Tracing, parsingErr = parseString(value)
+	case "tmpdirpath":
+		cfg.TmpDirPath, parsingErr = parseString(value)
+	case "disablequerycontextcache":
+		cfg.DisableQueryContextCache, parsingErr = parseBool(value)
+	case "includeretryreason":
+		cfg.IncludeRetryReason, parsingErr = parseConfigBool(value)
+	case "clientconfigfile":
+		cfg.ClientConfigFile, parsingErr = parseString(value)
+	case "disableconsolelogin":
+		cfg.DisableConsoleLogin, parsingErr = parseConfigBool(value)
+	case "disablesamlurlcheck":
+		cfg.DisableSamlURLCheck, parsingErr = parseConfigBool(value)
+	case "token_file_path":
+		tokenPath, parsingErr = parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		v, err := readToken(tokenPath)
+		if err != nil {
+			return err
+		}
+		cfg.Token = v
+	default:
+		param, parsingErr := parseString(value)
+		if err := checkParsingError(parsingErr, key, value); err != nil {
+			return err
+		}
+		cfg.Params[urlDecodeIfNeeded(key)] = &param
+	}
+	return checkParsingError(parsingErr, key, value)
+}
+
+func checkParsingError(parsingErr error, key string, value interface{}) error {
+	if parsingErr != nil {
+		err := &SnowflakeError{
+			Number:      ErrCodeTomlFileParsingFailed,
+			Message:     errMsgFailedToParseTomlFile,
+			MessageArgs: []interface{}{key, value},
+		}
+		return err
+	}
+	return nil
+}
+
+func parseInt(i interface{}) (int, error) {
+	v, ok := i.(string)
+	if !ok {
+		num, ok := i.(int)
+		if !ok {
+			return 0, errors.New("failed to parse the value to integer")
+		}
+		return num, nil
+	}
+	return strconv.Atoi(v)
+}
+
+func parseBool(i interface{}) (bool, error) {
+	v, ok := i.(string)
+	if !ok {
+		vv, ok := i.(bool)
+		if !ok {
+			return false, errors.New("failed to parse the value to boolean")
+		}
+		return vv, nil
+	}
+	return strconv.ParseBool(v)
+}
+
+func parseConfigBool(i interface{}) (ConfigBool, error) {
+	vv, err := parseBool(i)
+	if err != nil {
+		return ConfigBoolFalse, err
+	}
+	if vv {
+		return ConfigBoolTrue, nil
+	}
+	return ConfigBoolFalse, nil
+}
+
+func parseDuration(i interface{}) (time.Duration, error) {
+	v, ok := i.(string)
+	if !ok {
+		num, err := parseInt(i)
+		if err != nil {
+			return time.Duration(0), err
+		}
+		t := int64(num)
+		return time.Duration(t * int64(time.Second)), nil
+	}
+	return parseTimeout(v)
+}
+
+func readToken(tokenPath string) (string, error) {
+	if tokenPath == "" {
+		tokenPath = "./snowflake/session/token"
+	}
+	if !path.IsAbs(tokenPath) {
+		snowflakeConfigDir, err := getTomlFilePath(os.Getenv(snowflakeHome))
+		if err != nil {
+			return "", err
+		}
+		tokenPath = path.Join(snowflakeConfigDir, tokenPath)
```

Comment:
token file path shouldn't be set as a result of joining with snowflakeConfigDir. In specification we got the information that it should be either absolute path '/snowflake/session/token' or custom path set in configuration with key = 'token_file_path'

Reply:
Updated.